### PR TITLE
Reformatting IA for JTBD guides

### PR DESCRIPTION
**Pull request by renebrandel**
#### Description of changes:
- left hand nav updated according to JTBD guides
- renamed section headers for JTBD guides
- introduced an "all-javascript" filter to easily apply an InlineFilter to all javascript platforms.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/graphqlapi/modify-amplify-generated-resources/index.mdx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
 
 export const meta = {
-  title: 'Set up Amplify GraphQL API',
+  title: 'Modify Amplify-generated resources',
   description:
     "Learn more about how to get started with Amplify's API category",
   platforms: ['javascript', 'react-native', 'swift', 'android', 'flutter']
```
Comment (heatheramz) [inline src/pages/[platform]/build-a-backend/graphqlapi/modify-amplify-generated-resources/index.mdx:7]: @renebrandel This one looks like it was missed in the update as a couple things are diff from the new header

  platforms: [
                'android',
                'angular',
                'flutter',
                'javascript',
                'nextjs',
                'react',
                'react-native',
                'swift',
                'vue'
              ]
};

export const getStaticPaths = async () => {
  return getCustomStaticPath(meta.platforms);
};

export function getStaticProps(context) {
  return {
    props: {
      platform: context.params.platform,
      meta
    }
  };
}

File: src/pages/[platform]/build-a-backend/graphqlapi/search-and-result-aggregations/index.mdx [hunk 2]
```diff
@@ -0,0 +1,480 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Build search and aggregate queries',
+  description:
+    'Add authorization rules to your GraphQL schema to control access to your data.',
```
Comment (heatheramz) [inline src/pages/[platform]/build-a-backend/graphqlapi/search-and-result-aggregations/index.mdx:6]: Description and title don't match

File: src/pages/[platform]/build-a-backend/graphqlapi/custom-buisness-logic/index.mdx [hunk 3]
```diff
@@ -0,0 +1,871 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Set up custom queries and mutations',
+  description:
+    'Add authorization rules to your GraphQL schema to control access to your data.',
```
Comment (heatheramz) [inline src/pages/[platform]/build-a-backend/graphqlapi/custom-buisness-logic/index.mdx:6]: Title and description don't match

**Review by heatheramz — COMMENTED** [review]
@renebrandel a few small changes for your review - recommend looping in Scott early on the proposed changes to the JS fragment filters.

**Review by heatheramz — APPROVED** [review]
Thank you for updating LGTM but defer to the docs team on the inline filter update.

File: src/data/platforms.ts [hunk 4]
```diff
@@ -10,16 +10,20 @@ export type Platform =
...
   'react-native',
-  'swift',
   'vue'
+] as const;
+
+export const PLATFORMS: Platforms = [
+  'android',
+  'flutter',
+  'swift',
+  ...JS_PLATFORMS
```
Comment (reesscot) [inline src/data/platforms.ts:26]: good find

**Review by reesscot — APPROVED** [review]

**Pull request merged**